src/shared/cpu-set-util.c

```diff
@@ -82,6 +82,23 @@ char *cpu_set_to_range_string(const CPUSet *set) {
         return TAKE_PTR(str) ?: strdup("");
 }
 
+char* cpu_set_to_mask_string(const CPUSet *a) {
+        _cleanup_free_ char *str = NULL;
+        unsigned mask = 0;
+        int i, r;
+
+        for (i = 0; (size_t) i < a->allocated * 8; i++) {
+                if (!CPU_ISSET_S(i, a->allocated, a->set))
+                        continue;
+
+                mask += 1 << i;
+        }
+        r = asprintf(&str, "%x", mask);
+        assert_se(r > 0);
+
+        return TAKE_PTR(str) ?: strdup("");
```

Comment:
the ternary op is unnecessary. if asprintf() can fail, then strdup() can fail too, so why bother?